src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultApplicationModelProvider.cs

```diff
@@ -592,8 +589,7 @@ protected virtual bool IsAction(TypeInfo typeInfo, MethodInfo methodInfo)
             var routeTemplateProvider =
                 attributes
                 .OfType<IRouteTemplateProvider>()
-                .Where(a => !IsSilentRouteAttribute(a))
-                .SingleOrDefault();
+                .SingleOrDefault(a => !IsSilentRouteAttribute(a));
```

Comment:
That would hide a bug. It's intended that this method only be called when one or zero attributes is a non-silent route provided. TLDR all of this code is complicated on purpose.